skyportal/models/user_notification.py

```diff
@@ -638,32 +653,80 @@ def receive_after_flush(session, context):
+                                            raise ValueError(
+                                                f"Invalid propertiesFilter value: {e}"
+                                            )
+                                        op = prop_split[2].strip()
+                                        if op not in op_options:
+                                            raise ValueError(f"Invalid operator: {op}")
+                                        comp_function = getattr(operator, op)
+                                        if not comp_function(
+                                            localization_properties_dict[name], value
+                                        ):
+                                            send_notification = False
+                                            break
+
+                        if send_notification:
+                            stmt = sa.select(GcnNotice).where(
+                                GcnNotice.dateobs == target.dateobs
+                            )
+                            count_stmt = sa.select(func.count()).select_from(stmt)
+                            count_notices = session.execute(count_stmt).scalar()
+                            if count_notices > 1:
+                                session.add(
+                                    UserNotification(
+                                        user=user,
+                                        text=f"New Notice for GCN Event *{target.dateobs}*, with Notice Type *{gcn.NoticeType(notice.notice_type).name}*",
+                                        notification_type="gcn_events",
+                                        url=f"/gcn_events/{str(target.dateobs).replace(' ','T')}",
                                     )
-                                else:
-                                    session.add(
-                                        UserNotification(
-                                            user=user,
-                                            text=f"New GCN Event *{target.dateobs}*, with Notice Type *{gcn.NoticeType(target.notice_type).name}*",
-                                            notification_type="gcn_events",
-                                            url=f"/gcn_events/{str(target.dateobs).replace(' ','T')}",
-                                        )
+                                )
+                            else:
+                                session.add(
+                                    UserNotification(
+                                        user=user,
+                                        text=f"New GCN Event *{target.dateobs}*, with Notice Type *{gcn.NoticeType(notice.notice_type).name}*",
+                                        notification_type="gcn_events",
+                                        url=f"/gcn_events/{str(target.dateobs).replace(' ','T')}",
                                     )
+                                )
```

Comment:
Lets make the first few words of the `text` input inside the conditional and the rest of the code after it.

Reply:
good point.